be/btree.c

```diff
@@ -489,8 +489,8 @@ static void be_btree_split_child(struct m0_be_btree *btree,
 	i = 0;
 	while (i < new_child->bt_num_active_key) {
 		new_child->bt_kv_arr[i] = child->bt_kv_arr[i + BTREE_FAN_OUT];
-		new_child->bt_ik[i] = child->bt_ik[i + BTREE_FAN_OUT];
-		new_child->allocated[i] = child->allocated[i + BTREE_FAN_OUT];
+		*new_child->bt_ik[i].inlkey = *((char *)child->bt_kv_arr[i + BTREE_FAN_OUT].btree_key);
+		new_child->allocated[i] = 1;
 		if(btree->bb_ops->ko_type != M0_BBT_CAS_CTG)
```

Comment:
The check for M0_BBT_CAS_CTG should also cover the memcpy mentioned above. We do not want to do memcpy and also setting allocated array in case of M0_BBT_CAS_CTG.